eta/core/c3d.py

```diff
@@ -0,0 +1,340 @@
+'''
+Tensorflow implementation of the popular C3D network.
+
+This implementation extends and add functionality to the original C3D
+implementation by Hou Xin.
+
+C3D implementation in TensorFlow:
+https://github.com/hx173149/C3D-tensorflow
+Hou Xin, 2016
+
+Model weights:
+https://www.dropbox.com/sh/8wcjrcadx4r31ux/AAAkz3dQ706pPO8ZavrztRCca?dl=0
+
+Copyright 2018, Voxel51, LLC
+voxel51.com
+
+Yixin Jin, yixin@voxel51.com
+'''
+# pragma pylint: disable=redefined-builtin
+# pragma pylint: disable=unused-wildcard-import
+# pragma pylint: disable=wildcard-import
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+from builtins import *
+# pragma pylint: enable=redefined-builtin
+# pragma pylint: enable=unused-wildcard-import
+# pragma pylint: enable=wildcard-import
+
+import logging
+import math
+import os
+
+import numpy as np
+from sklearn.preprocessing import normalize
+import tensorflow as tf
+
+from eta.core.config import Config
+import eta.core.image as etai
+from eta.core.features import Featurizer
+import eta.core.video as etav
+
+
+logger = logging.getLogger(__name__)
+
+
+class C3DConfig(Config):
+    '''Configuration settings for the C3D network.'''
+
+    def __init__(self, d):
+         self.model = self.parse_string(d, "model", default="C3D")
```

Comment:
This is never used. I suspect you copied this from VGG-16 without understanding what it does. Don't copy code you don't understand!

Reply:
With that said, please do keep this field here, because you need to update the way the model is loaded here. While you were sick we introduced a model registry that allows us to "publish" models and refer to them by name.

Please refer to `vgg16.py` to see how this works. Basically all models will be storead Google Drive and your `_load_model` code here should use the `eta.core.models` tools to download the model if necessary and get the path to the model without you having to hard code it in `model_path` below.

Ask me for help on how to actually "publish the model".

Reply:
Removed model_path but tested with this config locally.